geoh5py/ui_json/utils.py

```diff
@@ -46,6 +46,12 @@ def flatten(ui_json: dict[str, dict]) -> dict[str, Any]:
 
                     if value.get("groupValue"):
                         temp_value = {"group": value["groupValue"], "value": temp_value}
+                    elif value.get("rangeLabel"):
```

Comment:
I guess I didn't recognize the significance of this in the first pass.  The output or flatten should be a flat (not nested) dictionary.  So temp_value here should just be value["value"] and then any code needing to access the state (property and isComplement) of the parameter should refer to the ui_json dictionary attribute

Reply:
If you fix this here, then you can also remove the dict handler in update_ui_values